build

```diff
@@ -176,21 +174,19 @@ def colorize(s):
 
 def pipe_errors_to_less_handler():
   global error_stream
-  if len (error_stream):
+  if error_stream:
     try:
       [ fid, name ] = tempfile.mkstemp()
       try:
         fid = os.fdopen (fid, 'wb')
-        [ fid.write (x.encode (errors='ignore')) for x in colorize(error_stream) ]
+        fid.write ([ x.encode (errors='ignore') for x in colorize(error_stream) ])
```

Comment:
OK, just noticed that this fails when there is output to be written - i.e. a compilation failure. Once I'd relaxed all the exception handling, the error was:
```
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "./build", line 181, in pipe_errors_to_less_handler
    fid.write ([ x.encode (errors='ignore') for x in colorize(error_stream) ])
TypeError: a bytes-like object is required, not 'list'
```
We'll need to revert that back...

Reply:
Should be fixed in #1412 - if & when it gets merged...

Reply:
One of my suggestions was going to be the solution you've got in 6d4f8383.

But I'm wondering if instead it would make more sense for `colorize()` to return a string rather than a list?

Reply:
Just had a look, and every usage of `colorize()` was a weird syntax specifically to deal with that function returning a list rather than a string. Changed in #1412, diff of bcad65e should demonstrate what I mean.

Reply:
👍